client/asset/btc/btc.go

```diff
@@ -658,22 +664,23 @@ out:
 	}
 
 	if btc.useSplitTx && !ord.Immediate {
-		return btc.split(ord.Value, ord.MaxSwapCount, spents, uint64(size), fundingCoins, ord.DEXConfig)
+		coins, err = btc.split(ord.Value, ord.MaxSwapCount, spents, uint64(size), fundingCoins, ord.DEXConfig)
+		return coins, nil, err // no redeem script required for split tx output
```

Comment:
Aha. This could have been problematic if `btc.split` returned the original funding coins. If that were to happen, `redeemScripts` would need to be returned here.

Reply:
As I mentioned in https://github.com/decred/dcrdex/pull/679#discussion_r487437971, I worry that these kinds of bugs could easily slip in with the current implementations. I'll be pushing a refactor soon and it would make review a little tougher but the code flow should be _easier_ to follow and thus _easier_ to spot bugs.

Reply:
All for improvements to readability, but like @buck54321 I want to go easy on refactoring.  Partly to get these fixes through quickly, but also because it makes review harder and inevitably introduces bugs instead.

Reply:
> I want to go easy on refactoring. Partly to get these fixes through quickly, but also because it makes review harder and inevitably introduces bugs instead.

I'll try to keep the refactors in this PR smaller so it can go in faster but I'd gone somewhat far in the refactoring efforts and still see feel the refactors would be very helpful. The refactoring process has helped me to discover some bug sites that either already existed or would creep in with this PR (such as this one). I'll mention some of such existing issues in a review below to illustrate why simply fixing the observed issues will only lead to more code entropy and make these code more difficult to maintain in subsequent work.

Reply:
> I want to go easy on refactoring. Partly to get these fixes through quickly, but also because it makes review harder and inevitably introduces bugs instead.

I understand. I do believe the refactor will help prevent bugs rather than introduce them as I've found some tricky spots withe current implementation that I've been able to address in the refactor. Plus getting FundOrder to return a `struct{Coin, dex.Bytes}`  complicates the code a lot more without slipping in some refactors. I think the key changes with the refactor are:
- adding a `asset.OrderCoin` interface that extends the `Coin` interface with a `Redeem()` method, to allow room for asset-specific implementations
- having `compositeUTXO` satisfy the new `asset.OrderCoin` interface
- and more importantly, make dcr and btc implementations much more similar.

Reply:
I recognized that if I do not try to make FundOrder return `[]struct{Coin, dex.Bytes}`, I can keep the changes to this PR fewer and get it in faster. I'll open a follow up PR for the refactor with the above key changes.